src/main/java/teammates/logic/core/GoogleCloudLoggingService.java

```diff
@@ -94,4 +113,127 @@
+        LogEntry entry = LogEntry.newBuilder(StringPayload.of(payload))
+                .setLogName(FEEDBACK_SESSION_LOG_NAME)
+                .addLabel(FEEDBACK_SESSION_LOG_COURSE_ID_LABEL, courseId)
+                .addLabel(FEEDBACK_SESSION_LOG_EMAIL_LABEL, email)
+                .addLabel(FEEDBACK_SESSION_LOG_NAME_LABEL, fsName)
+                .addLabel(FEEDBACK_SESSION_LOG_TYPE_LABEL, fslType)
+                .setResource(MonitoredResource.newBuilder("global").build())
+                .build();
+        createLogEntry(entry);
+    }
+
+    private void createLogEntry(LogEntry entry) throws LogServiceException {
+        try (Logging logging = LoggingOptions.getDefaultInstance().getService()) {
+            logging.write(Collections.singleton(entry));
+            logging.close();
+        } catch (Exception e) {
+            throw new LogServiceException(e);
+        }
+    }
+
+    @Override
+    public List<FeedbackSessionLogEntry> getFeedbackSessionLogs(String courseId, String email,
+            Instant startTime, Instant endTime) throws LogServiceException {
+        LogSearchParams logSearchParams = new LogSearchParams()
+                .setLogName(FEEDBACK_SESSION_LOG_NAME)
+                .addLabel(FEEDBACK_SESSION_LOG_COURSE_ID_LABEL, courseId)
+                .addLabel(FEEDBACK_SESSION_LOG_EMAIL_LABEL, email)
+                .setStartTime(startTime)
+                .setEndTime(endTime);
+        List<LogEntry> logEntries = getLogEntries(logSearchParams);
+
+        List<FeedbackSessionLogEntry> fsLogEntries = new ArrayList<>();
+        for (LogEntry entry : logEntries) {
+            String fslType = entry.getLabels().get(FEEDBACK_SESSION_LOG_TYPE_LABEL);
+            long timestamp = entry.getTimestamp();
+            String entryEmail = entry.getLabels().get(FEEDBACK_SESSION_LOG_EMAIL_LABEL);
+            String entryFsName = entry.getLabels().get(FEEDBACK_SESSION_LOG_NAME_LABEL);
+            StudentAttributes student = studentsLogic.getStudentForEmail(courseId, entryEmail);
+            FeedbackSessionAttributes fs = fsLogic.getFeedbackSession(entryFsName, courseId);
+            if (student == null || fs == null) {
+                // If the student email or feedback session retrieved from the logs are invalid, discard it
+                continue;
+            }
+            if (!fslType.equals(Const.FeedbackSessionLogTypes.ACCESS)
+                    && !fslType.equals(Const.FeedbackSessionLogTypes.SUBMISSION)) {
+                // If the feedback session log type retrieved from the logs is invalid, discard it
+                continue;
+            }
+            FeedbackSessionLogEntry fslEntry = new FeedbackSessionLogEntry(student, fs, fslType, timestamp);
+            fsLogEntries.add(fslEntry);
+        }
+
+        return fsLogEntries;
+    }
+
+    private List<LogEntry> getLogEntries(LogSearchParams s) throws LogServiceException {
```

Comment:
You can probably refactor the existing `getRecentErrorLogs` method to also make use of this.

Reply:
Should this be done separately on the master branch instead? It can be done here too, but was a bit hesitant cos this branch is specifically for the audit logs feature